src/GraphQL/Resolvers/SourceStreamMethodResolver.cs

```diff
@@ -61,10 +65,20 @@ public SourceStreamMethodResolver(MethodInfo methodInfo, LambdaExpression instan
                         taskBodyExpression = Expression.Call(_castFromTaskAsyncMethodInfo.MakeGenericMethod(innerType), bodyExpression);
                     }
                 }
+                // Task<IAsyncEnumerable<T>>
+                else if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(IAsyncEnumerable<>))
+                {
+                    var innerType = type.GetGenericArguments()[0];
+                    var method = _convertFromTaskAsyncEnumerableMethodInfo.MakeGenericMethod(innerType);
+                    var func = method.CreateDelegate<Func<Expression, ParameterExpression, Func<IResolveFieldContext, ValueTask<IObservable<object?>>>>>();
+                    return func(bodyExpression, resolveFieldContextParameter);
```

Comment:
Thought about design. As you said `SourceStreamMethodResolver` was already complicated and now has become event more complicated. Thought each if-branch individually is simple. What I don't like here - a mix of return statements in individual (new) branches vs setting `taskBodyExpression` with returning result at the very end. I would switch to return in the each if-branch and remove `else`.  So only `throw new InvalidOperationException` would stay at the very end.

Reply:
Ok done.